pkg/datapath/linux/ipsec/xfrm_collector.go

```diff
@@ -37,32 +37,42 @@ const (
 )
 
 type xfrmCollector struct {
-	xfrmStatFunc func() (procfs.XfrmStat, error)
+	xfrmStatFunc          func() (procfs.XfrmStat, error)
+	maxSequenceNumberFunc func() (uint32, error)
 
 	// Inbound errors
 	xfrmErrorDesc *prometheus.Desc
+	//Max seq.number
+	maxSequenceNumberDesc *prometheus.Desc
 }
 
 // NewXFRMCollector returns a new prometheus.Collector for /proc/net/xfrm_stat
 // https://www.kernel.org/doc/Documentation/networking/xfrm_proc.txt
 func NewXFRMCollector() prometheus.Collector {
-	return newXFRMCollector(procfs.NewXfrmStat)
+	return newXFRMCollector(procfs.NewXfrmStat, maxSequenceNumber)
 }
 
-func newXFRMCollector(statFn func() (procfs.XfrmStat, error)) prometheus.Collector {
+func newXFRMCollector(statFn func() (procfs.XfrmStat, error), maxSeqNumberFn func() (uint32, error)) prometheus.Collector {
 	return &xfrmCollector{
-		xfrmStatFunc: statFn,
+		xfrmStatFunc:          statFn,
+		maxSequenceNumberFunc: maxSeqNumberFn,
 
 		xfrmErrorDesc: prometheus.NewDesc(
 			prometheus.BuildFQName(metrics.Namespace, subsystem, "xfrm_error"),
 			"Total number of xfrm errors",
 			[]string{labelErrorType, metrics.LabelError}, nil,
 		),
+		maxSequenceNumberDesc: prometheus.NewDesc(
+			prometheus.BuildFQName(metrics.Namespace, subsystem, "xfrm_max_sequence_number"),
+			"Max sequence number of xfrm anti-replay",
```

Comment:
Let's maybe call it "Highest" here to not give the impression it's the actual limit:
```suggestion
"Highest XFRM anti-replay sequence number on the node",
```

Reply:
done